coordinator/src/services/validation/fhir-path.ts

```diff
@@ -0,0 +1,61 @@
+import {Bundle, Extension, Reference, Resource} from "../../models/fhir/fhir-resources"
+import {resolveReference} from "../translation/common"
+
+/**
+ * Implementation of simple FHIR paths
+ * See https://www.hl7.org/fhir/fhirpath.html#simple
+ */
+
+export function applyFhirPath(bundle: Bundle, input: Array<unknown>, path: string): Array<unknown> {
+  const pathElements = splitFhirPath(path)
+  return pathElements.reduce(
+    (output, pathElement) => applyFhirPathElement(bundle, output, pathElement),
+    input
+  )
+}
+
+function splitFhirPath(path: string) {
```

Comment:
This function takes a FHIR path, e.g.:
`entry.resource.ofType(MedicationRequest).dispenseRequest.extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType")`
and returns the individual elements, e.g.:
* `entry`
* `resource`
* `ofType(MedicationRequest)`
* `dispenseRequest`
* `extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType")`

Note we can't just split on `.` because the `extension()` elements contain URLs.

This is a fairly naive implementation - it doesn't for example deal with escaped quotes. However we only ever parse FHIR paths which we have written, never ones from a user, so this doesn't have to be bulletproof.

Reply:
But then we'd have two problems

Reply:
Very clever 😉 Happy to go either way, but all things being equal I prefer having less code rather than more

Reply:
Nah I was kidding. Didn't occur to me to use regex

Reply:
Although I'm going to need to convince myself that that regex does what I want it do. Which is the joy of using regex instead of code :unamused: